front/lib/actions/mcp_internal_actions/servers/schedules_management.ts

```diff
@@ -0,0 +1,502 @@
+          workspaceId: owner.id,
+          agentConfigurationId: agentConfiguration.sId,
+          name,
+          kind: "schedule",
+          enabled: true,
+          configuration: {
+            cron,
+            timezone: resultTimezone,
+          },
+          naturalLanguageDescription: schedule,
+          customPrompt: prompt,
+          editor: user.id,
+          webhookSourceViewId: null,
+          executionPerDayLimitOverride: null,
+          executionMode: "fair_use",
+        });
+
+        if (result.isErr()) {
+          return new Err(
+            new MCPError(`Failed to create schedule: ${result.error.message}`)
+          );
+        }
+
+        getStatsDClient().increment("tools.schedules_management.created", 1, [
+          `workspace_id:${owner.sId}`,
+          `agent_id:${agentConfiguration.sId}`,
+        ]);
+
+        return new Ok([
+          {
+            type: "text" as const,
+            text:
+              `Created schedule "${name}"!\n\n` +
+              `Schedule: ${schedule}\n` +
+              `Cron: ${cron} (${resultTimezone})\n\n` +
+              `The agent will execute "${prompt}" according to this schedule.\n\n` +
+              renderSchedule(result.value),
+          },
+        ]);
+      }
+    )
+  );
+
+  server.tool(
+    "list_schedules",
+    "List all schedules created for this agent.",
+    {},
+    withToolLogging(
+      auth,
+      {
+        toolNameForMonitoring: "schedules_management_list",
+        agentLoopContext,
+      },
+      async () => {
+        const contextResult = getToolContext(auth, agentLoopContext);
+        if (contextResult.isErr()) {
+          return contextResult;
+        }
+        const { userId, workspaceId, agentConfiguration } = contextResult.value;
+
+        const schedulesResult =
+          await TriggerResource.listSchedulesByAgentAndEditor(auth, {
+            agentConfigurationId: agentConfiguration.sId,
+            editorIds: [userId],
+          });
+
+        if (schedulesResult.isErr()) {
+          return new Err(
+            new MCPError("Error while fetching schedules for this agent")
+          );
+        }
+
+        getStatsDClient().increment("tools.schedules_management.listed", 1, [
+          `workspace_id:${workspaceId}`,
+          `agent_id:${agentConfiguration.sId}`,
+        ]);
+
+        if (schedulesResult.value.length === 0) {
+          return new Ok([
+            {
+              type: "text" as const,
+              text: "No schedules configured for this agent.",
+            },
+          ]);
+        }
+
+        const scheduleList = schedulesResult.value
+          .map((schedule) => renderSchedule(schedule))
+          .join("\n\n");
+
+        return new Ok([
+          {
+            type: "text" as const,
+            text: `Schedules for this agent:\n\n${scheduleList}`,
+          },
+        ]);
+      }
+    )
+  );
+
+  server.tool(
+    "get_schedule",
```

Comment:
why do we have a singular get schedule by ID ? we already have all the same data in the tool above (list schedules), why/how would an agent query the specific id without listing first, and why would it be useful because there's the same data ?

i think we can cut this, the less tool overwhelm for the agent the better imho